components/popupSearch/index.tsx

```diff
@@ -0,0 +1,91 @@
+import React, {useEffect, useState} from "react";
+import {
+  CloseIconStyles,
+  Input,
+  InputContainer,
+  Popup,
+  PopupBackground,
+  SearchListContainer
+} from "./popupSearch.styled";
+import {SearchItem} from "../search/SearchItem";
+import CloseIcon from '../../assets/icons/closeIcon.svg'
+import {SearchErrorMessage} from "../search/SearchErrorMessage";
+import {HeaderProps} from "../../interfaces/header";
+import {PopupSearchItemType} from "../../interfaces/popupSearch";
+
+export const PopupSearch: React.FC<HeaderProps> = ({setIsPopup, isPopup}) => {
+  const [searchText, setSearchText] = useState('')
+  const [searchList, setSearchList] = useState<PopupSearchItemType[] | []>([])
+
+  const fetchApi = async (searchText: string) => {
+    try {
+      const res = await fetch(`https://api.jikan.moe/v4/anime?limit=5&letter=${searchText}`)
+      const {data} = await res.json()
+      setSearchList(data)
+    } catch (e) {
+      setSearchList([])
+      console.log('e', e)
+    }
+  }
+
+  useEffect(() => {
+
+    if (searchText.length >= 2) {
+      fetchApi(searchText)
+    }
+
+  }, [searchText])
+
+
+  const handleValueInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    
+    let value = e.target.value.trim()
+
+    setSearchText(value)
```

Comment:
what is the sense of setting search text here and one more time here https://github.com/emfree1/Anime-Next-site/pull/1/files#diff-1c6f30962151ebb60625b41756b2fd7b28ace25587c0cceafe4281f9490a04b1R47 in same function?
I would move if (!value) condition to the top of the function and get read of calling it 2 times